src/enum.cr

```diff
@@ -311,8 +311,11 @@ struct Enum
   # ```
   def self.from_value?(value) : self | Nil
     {% if @type.has_attribute?("Flags") %}
-      arr = values.select {|v| (v.to_i & value) == v.to_i && v.to_i != 0 }
-      return arr.reduce { |types, e| types | e } unless arr.empty?
+      mask = {% for member, i in @type.constants %}\
+        {% if i != 0 %} | {% end %}\
```

Comment:
Yes, but in macros the alternative is to join everything in a single line and it is less readable IMO.
Since the macro has test coverage I think we are fine here.